### PR TITLE
[SYCLomatic] Fix the OOM issue when migrate too many files together

### PR DESCRIPTION
**Pull request by ziranzha**
Signed-off-by: Ziran Zhang <ziran.zhang@intel.com>

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/MigrationAction.cpp [hunk 1]
```diff
@@ -14,11 +14,67 @@
...
+constexpr size_t MinAvailableMemorySize = 512 * 1024 * 1024; // 512Mb
+constexpr size_t MinAvailableMemoryPercent = 25;             // 25 percent
+
+/// Return true if available phy memory is larger than \p
+/// MinAvailableMemorySize and available phy memory percents is larger than \p
+/// MinAvailableMemoryPercent. Otherwise return false.
+bool checkMemoryStatus();
+
+#ifdef _WIN32
+bool checkMemoryStatus() {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MigrationAction.cpp]: canUseMoreMemory() ,  true:  yes we can use more.

File: clang/lib/DPCT/MigrationAction.cpp [hunk 2]
```diff
@@ -14,11 +14,67 @@
...
+  MEMORYSTATUSEX MStatus;
+  MStatus.dwLength = sizeof(MStatus);
+  GlobalMemoryStatusEx(&MStatus);
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MigrationAction.cpp]: Please check the retrun value of 41,  and if it fails return a predefined default behaiour.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**